src/bin/ion/commands/generate/generator.rs

```diff
@@ -270,39 +382,97 @@ impl<'a, L: Language + 'static> CodeGenerator<'a, L> {
         isl_type: &IslType,
     ) -> CodeGenResult<()> {
         let mut context = Context::new();
-        let mut tera_fields = vec![];
         let mut code_gen_context = CodeGenContext::new();
-        let mut nested_types = vec![];
 
-        // Set the ISL type name for the generated abstract data type
-        context.insert("target_kind_name", &isl_type_name.to_case(Case::UpperCamel));
+        code_gen_context.with_data_model_node(DataModelNode {
+            name: isl_type_name.to_case(Case::UpperCamel),
+            code_gen_type: None,
+            nested_types: vec![],
+        });
+
+        if let Some(ref mut fully_qualified_type_name) = self.current_type_fully_qualified_name {
+            fully_qualified_type_name.push(isl_type_name.to_case(Case::UpperCamel));
+        }
```

Comment:
> because Rust doesn't need it

Rust code-gen should have fully qualified names. In generated code, it's best to use fully qualified references such as `my_model::common::Foo` to avoid unexpected name clashes.

Reply:
`self.current_type_fully_qualified_name` is initialized when the code generator itself is initialized with a namespace. Although currently Rust code generator sets it as `None` but once I add back support for Rust, it will be `Some(NAMESPACE)`. Like Matthew said, this fully qualified type name is going to be used by the model, so it would be used by both Rust and Java.

Reply:
> Although currently Rust code generator sets it as `None`

That seems odd to me. There should be a language agnostic layer that reads ISL and builds this data model. Then, individual code generators can read this model and produce code for the target language.